lib/service-manager.js

```diff
@@ -334,11 +334,15 @@ prototype.getInstanceMetas = function(callback) {
     for (var i in instances) {
       if (!instances.hasOwnProperty(i)) continue;
       var instance = instances[i];
-      debug('%s: %j', iid(instance), instance.containerVersionInfo);
-      metas[instance.id] = instance.containerVersionInfo;
+      var meta = instance.containerVersionInfo;
+      debug('%s: meta %j', iid(instance), meta);
 
-      debug('%s: size %j', iid(instance), instance.cpus);
-      metas[instance.id].size = instance.cpus;
+      if (meta) {
```

Comment:
If you look again, you will see in the absence of containerInfoVersion, that no meta for this instance is inserted into the `metas` object. In other words, an instance without containerVersionInfo is not considered restartable, and never gets passed to the driver via `driver.start(metas)`, so the driver needs no such check.